src/App.tsx

```diff
@@ -1,24 +1,246 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, { useEffect, useMemo, useState } from 'react';
+import { UserWarning } from './components/UserWarning/UserWarning';
+import { client } from './utils/fetchClient';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList/TodoList';
+import { Header } from './components/Header/Header';
+import { Footer } from './components/Footer/Footer';
+import { Errors } from './components/Errors/Errors';
+import { Status } from './types/Status';
+import { ErrorSpec } from './types/ErrorSpec';
+import { filterTodos } from './helpers/filterTodos';
+import * as TodoService from './services/todo';
 
-const USER_ID = 0;
+const USER_ID = 12021;
+const ADDED_URL = `/todos?userId=${USER_ID}`;
 
 export const App: React.FC = () => {
+  const [todosFromServer, setTodosFromServer] = useState<Todo[]>([]);
+  const [hasErrors, setHasErrors] = useState(false);
+  const [error, setError] = useState<ErrorSpec | null>(null);
+  const [status, setStatus] = useState<Status>(Status.ALL);
+  const [uncompletedTodosCount, setUncompletedTodosCount] = useState<number>(0);
+  const [isInputDisabled, setIsInputDisabled] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [loadingTodo, setLoadingTodo] = useState<Todo | null>(null);
+  const [editingTodo, setEditingTodo] = useState<Todo | null>(null);
+
+  const areAllCompleted = uncompletedTodosCount === 0;
+
+  const filteredTodos = useMemo(
+    () => filterTodos(todosFromServer, status),
+    [status, todosFromServer],
+  );
+
+  useEffect(() => {
+    client
+      .get<Todo[]>(ADDED_URL)
+      .then((todos) => {
+        setTodosFromServer(todos);
+
+        if (todos.length === 0) {
```

Comment:
if the server responded without an error and returned an empty array of entities, this is not an error